ui/output.go

```diff
@@ -0,0 +1,91 @@
+package ui
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io"
+	"io/ioutil"
+	"strings"
+)
+
+// OutPutter handles output types
+type OutPutter interface {
+	Output() error
+}
+
+// Throughput reads from the from source, and sends to the to destination
+type Throughput struct {
+	from io.ReadCloser
+	to   io.Writer
+}
+
+// JSONThroughputter reads from the from source and sends to the to destination in JSON encoding
+type JSONThroughputter struct {
+	Throughput
+	pretty bool
+}
+
+// PlainThroughputter reads from the from source, and sends to the destination. The output is determined by the template and the data is read into the
+// dataStructure
+type PlainThroughputter struct {
+	Throughput
+	template      string
+	dataStructure interface{}
+}
+
+// NewOutPutter returns a configured OutPutter
+func NewOutPutter(format string, from io.ReadCloser, to io.Writer, template string, dataStructure interface{}) OutPutter {
+	switch strings.ToLower(format) {
+	case "json":
+		return &JSONThroughputter{Throughput: Throughput{from: from, to: to}, pretty: true}
+	default:
+		return &PlainThroughputter{Throughput: Throughput{from: from, to: to}, template: template, dataStructure: dataStructure}
+	}
+}
+
+// PrettyPrint enables or disables pretty json output
+func (j *JSONThroughputter) PrettyPrint(enabled bool) {
```

Comment:
Actually, looking at this again, I think this will never be used, so I'll just remove it.